xsuite/src/proxy/proxy.ts

```diff
@@ -29,22 +29,38 @@ export class Proxy {
     this.blockNonce = params.blockNonce ?? null;
   }
 
-  fetchRaw(path: string, data?: any) {
-    const baseUrl = this.proxyUrl;
-    const init: RequestInit = { headers: this.headers };
-    const url = new URL(`${baseUrl}${path}`);
-    if (this.blockNonce !== null) {
-      url.searchParams.append("blockNonce", this.blockNonce.toString());
+  private makeUrl(
+    path: string | URL,
+    params: Record<string, { toString: () => string } | undefined | null> = {},
+    override: boolean = false,
+  ) {
+    const url = new URL(path, this.proxyUrl);
+    for (const [k, v] of Object.entries(params)) {
+      if (v !== undefined && v !== null) {
+        if (override) {
+          url.searchParams.set(k, v.toString());
+        } else {
+          url.searchParams.append(k, v.toString());
+        }
+      }
     }
+    return url;
+  }
+
+  fetchRaw(url: URL, data?: any) {
+    const init: RequestInit = { headers: this.headers };
     if (data !== undefined) {
       init.method = "POST";
       init.body = JSON.stringify(data);
     }
-    return fetch(url.toString(), init).then((r) => r.json());
+    return fetch(url, init).then((r) => r.json());
   }
 
-  async fetch(path: string, data?: any) {
-    const res = await this.fetchRaw(path, data);
+  async fetch(url: string | URL, data?: any) {
```

Comment:
Is there a need for developers to allow the type URL there?

Reply:
It's here because `makeUrl` returns an `URL` object and as I said before, its to stay with the same object during all the flow

Reply:
If there is no need coming from developers, then we should not have `URL` as a type for this parameter. The reason is that the more we expose or allow, the harder it will be in the future to change stuff.